### PR TITLE
Feature/filtering

### PR DESCRIPTION
**Pull request by jmzwar**
Screen Shot 2023-05-22 at 9 24 20 am
Markets working - size slightly changed (removed $) to filter size just on asset size.

### CODE REVIEW

File: v2/perps-v2/ui/src/hooks/useActions.ts [hunk 1]
```diff
@@ -155,7 +181,9 @@ export const useActions = (account?: string, limit?: number) => {
       orderDirection: OrderDirection.Desc,
       where: {
         trader: accountLower,
-        market: marketAddress,
+        market_in: markets,
+        size_gte: min ? wei(min).toBN().toString() : undefined, // Should be divided by current price
+        size_lte: max ? wei(max).toBN().toString() : undefined, // Should be divided by current price
```
Comment (bachstatter) [inline v2/perps-v2/ui/src/hooks/useActions.ts:186]: What does should be divided by current mean?

**Comment by bachstatter** [comment]
Size is in native units? Would be nice to mention that.. Or maybe update the graph so we can use dollar size filter.. As of now you cant really use the size filter for all markets
Screenshot 2023-05-22 at 10 28 23 am

**Comment by bachstatter** [comment]
Selected markets not showing up and the popup is closed, is that expected?
Screenshot 2023-05-22 at 10 28 16 am
Screenshot 2023-05-22 at 10 30 48 am

**Comment by jaderns** [comment]
Supporting Joey's feedback. 
Can we make both filters apply when click? As in results reload as you select a market and as you type in the value for size? 
Size filter should ideally be in dollar unit
When popup closed, selected markets should be displayed (see ss)
Screenshot 2023-05-22 at 10 48 42

File: v2/perps-v2/ui/src/hooks/useActions.ts (hunk 1 above)
Comment (jmzwar) [reply v2/perps-v2/ui/src/hooks/useActions.ts:186]: This means the current price - but yeah I went with native assets to make it easier.

File: v2/perps-v2/ui/src/components/Shared/MarketSelect/MarketSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            <MarketCheckbox
+              key={market}
+              icon={<CurrencyIcon width="20px" height="20px" currencyKey={market} />}
+              label={market}
+              onChange={(e) => {
+                if (activeAssets.includes(market) && !e.target.checked) {
+                  const newState = activeAssets.filter((asset) => asset !== market);
+                  setActiveAssets(newState);
+                } else {
+                  setActiveAssets([...activeAssets, market]);
```
[the remark this answers is not in the document]
Comment (jmzwar) [reply v2/perps-v2/ui/src/components/Shared/MarketSelect/MarketSelect.tsx]: Hmm I'm not too worried about the query string. Results are sorted ofc by timestamp

File: v2/perps-v2/ui/src/utils/constants.ts [hunk 3]
```diff
@@ -102,6 +103,7 @@ export interface FuturesMarketConfig {
 export const MARKETS: Record<FuturesMarketKey, FuturesMarketConfig> = {
   // perps v2
   [FuturesMarketKey.sETHPERP]: {
+    id: '0x2b3bb4c683bfc5239b029131eef3b1d214478d93',
```
[the remark this answers is not in the document]
Comment (jmzwar) [reply v2/perps-v2/ui/src/utils/constants.ts:106]: Yeah true but too lazy now

**Comment by bachstatter** [comment]
Screenshot 2023-05-25 at 10 49 37 am
Screenshot 2023-05-25 at 10 49 30 am
Filter doesn't seem to work for me?

**Comment by bachstatter** [comment]
Screenshot 2023-05-25 at 10 50 00 am
Size not showing up in input field

**Review by bachstatter — CHANGES_REQUESTED** [review]
See comments

**Review by bachstatter — APPROVED** [review]
I pushed up one commit to ensure min/max gets set from query params. Nikita's comments makes sense too..

**Pull request merged**